### PR TITLE
Spacecat 634 jh2 vm benchmarktest

### PR DESCRIPTION
**Pull request by fraugabel**

### CODE REVIEW

File: features/steps/tools.py [hunk 1]
```diff
@@ -173,15 +173,35 @@ def check_volumes_created(client, test_name):
...
     ips = []
-    for port in ports:
-        for fixed_ip in port.fixed_ips:
-            ips.append(fixed_ip["ip_address"])
-    return ips
+    assertline=None
+    floating_ips = client.network.ips()
+    for ip in floating_ips:
+        ips.append(ip.floating_ip_address)
+        logger.log_info(f"found {ip.floating_ip_address}")
```
Comment (piobig2871) [inline features/steps/tools.py:182]: fix pep

File: features/openstack_testflow.feature [hunk 2]
```diff
@@ -3,25 +3,25 @@ Feature: Test creation and deletion of resources
...
     Given I connect to OpenStack
 
-    Then I should be able to create <router_quantity> routers
-    Then I should be able to create <network_quantity> networks
-    Then I should be able to create <subnet_quantity> subnets
-    Then I should be able to create <security_group_quantity> security groups
-    Then I should be able to create <security_group_rules_quantity> security group rules
-    Then I create <quantity_volumes> volumes
-
+   # Then I should be able to create <router_quantity> routers
```
Comment (piobig2871) [inline features/openstack_testflow.feature:6]: I don't want to merge this kind of commented files, that is not working

File: features/environment.py [hunk 3]
```diff
@@ -45,6 +46,9 @@ def before_all(context):
 
     context.collector = Collector()
 
+    # userdata = context.config.userdata
```
Comment (piobig2871) [inline features/environment.py:49]: why is it in before all

File: features/openstack_benchmark_pingVM.feature [hunk 4]
```diff
@@ -4,17 +4,16 @@ Feature: Benchmark ping VMs
...
-    Then I should be able to collect all VM IPs
-    And be able to ping all IPs 
-    # Then I should be able to access <jh_quantity> VMs
-    # Then all VMs should be able to ping each other
-    # TODO: Connectivity through jump hosts isn't implemented.
-    #  For this we need redirection rules on those jump hosts VMs which address our VMs.
+    #Given I have deployed a JH with the name <jh_name>
+    # TODO:
+    Given I have deployed <jh_quantity> JHs
+    and I have a private key at <vm_private_ssh_key_path> for <username>
```
Comment (piobig2871) [inline features/openstack_benchmark_pingVM.feature:10]: not finished

File: features/steps/definitions.py [hunk 5]
```diff
@@ -529,20 +529,46 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
...
-    def test_ssh_connection(context, username):
-        ssh_client = SshClient(context.vm_ip_address, username, context.vm_private_ssh_key_path)
+    
+    @then ("I should be able to SSH into {jh_quantity:d} JHs and test their {conn_test} connectivity")
+    def step_iterate_steps(context, jh_quantity,conn_test:str):
+        context.assertline=None
+        for i in range(1, jh_quantity + 1):
+            if not isinstance(context.jh,str):
+                context.vm_ip_address=context.jh[i-1]['ip']            
+                #print(f"ip {1}: {context.vm_ip_address}, {conn_test} ")
```
Comment (piobig2871) [inline features/steps/definitions.py:556]: remove print

File: features/steps/definitions.py [hunk 6]
```diff
@@ -529,20 +529,46 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
...
-    @then("I should be able to SSH into the VM as user {username}")
-    def test_ssh_connection(context, username):
-        ssh_client = SshClient(context.vm_ip_address, username, context.vm_private_ssh_key_path)
+    
+    @then ("I should be able to SSH into {jh_quantity:d} JHs and test their {conn_test} connectivity")
+    def step_iterate_steps(context, jh_quantity,conn_test:str):
+        context.assertline=None
+        for i in range(1, jh_quantity + 1):
+            if not isinstance(context.jh,str):
+                context.vm_ip_address=context.jh[i-1]['ip']
```
Comment (piobig2871) [inline features/steps/definitions.py:555]: why are you using i-1? why do you want to start counting from 0, in that case change the range from (0, jh_quantity)

File: features/steps/definitions.py [hunk 7]
```diff
@@ -529,20 +529,46 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
...
+                context.vm_ip_address=context.jh[i-1]['ip']            
+                #print(f"ip {1}: {context.vm_ip_address}, {conn_test} ")           
+                context.execute_steps('''
+                    Then I should be able to SSH into the VM
+                    Then I should be able to collect all VM IPs
+                    And be able to ping all IPs to test {conn_test} connectivity
+                    ''')
+            else:
+                context.assertline = f"No matching Jumphosts was found"
+                print(f"assert {context.assertline}")
```
Comment (piobig2871) [inline features/steps/definitions.py:564]: remove print please

File: features/steps/definitions.py [hunk 8]
```diff
@@ -552,19 +578,22 @@ def test_internet_connectivity(context):
...
-        tot_ips = len(context.ips)
-        for ip in context.ips:
-            result, assertline = context.ssh_client.test_internet_connectivity(ip, tot_ips)
-        print(result)
-        assert result[1] == 0, assertline
-
-    @then("be able to communicate with {ip}")
-    def test_domain_connectivity(context, ip: str):
-        result, assertline = context.ssh_client.test_internet_connectivity(ip)
+        context.ips,assertline=tools.collect_ips(context.client, context.logger)
```
Comment (piobig2871) [inline features/steps/definitions.py:581]: fix pep

File: features/steps/definitions.py [hunk 9]
```diff
@@ -552,19 +578,22 @@ def test_internet_connectivity(context):
...
+        if assertline != None:
+            context.assertline = assertline
+    
+    @then("be able to ping all IPs to test {conn_test} connectivity") 
+    def ping_ips_test(context, conn_test: str):
+        tot_ips=len(context.ips)
```
Comment (piobig2871) [inline features/steps/definitions.py:587]: add spaces next to "="

File: features/steps/definitions.py [hunk 10]
```diff
@@ -552,19 +578,22 @@ def test_internet_connectivity(context):
...
+        if len(context.ips) > 0:
+            for ip in context.ips:
+                result,assertline=context.ssh_client.test_internet_connectivity(conn_test, ip,tot_ips)
+            if result[1] != 0:
+                context.assertline = assertline
+
+    @then("be able to communicate with {ip} to test {conn_test} connectivity")
+    def test_domain_connectivity(context, ip: str, conn_test: str):       
+        result,assertline=context.ssh_client.test_internet_connectivity(conn_test, ip)
```
Comment (piobig2871) [inline features/steps/definitions.py:596]: as above

File: features/steps/tools.py [hunk 11]
```diff
@@ -173,15 +173,35 @@ def check_volumes_created(client, test_name):
...
-def collect_ips(client):
-    print("collecting ips")
-    ports = client.network.ports()
+def collect_ips(client,logger:Logger):
     ips = []
-    for port in ports:
-        for fixed_ip in port.fixed_ips:
-            ips.append(fixed_ip["ip_address"])
-    return ips
+    assertline=None
```
Comment (piobig2871) [inline features/steps/tools.py:178]: as above

File: features/steps/tools.py [hunk 12]
```diff
@@ -173,15 +173,35 @@ def check_volumes_created(client, test_name):
...
+    for ip in floating_ips:
+        ips.append(ip.floating_ip_address)
+        logger.log_info(f"found {ip.floating_ip_address}")
+    if len(ips)==0:
+        assertline= f"No ips found"
+    return ips, assertline
+
+def collect_jhs(client, test_name,logger:Logger):
+    servers=client.compute.servers()
+    lookup= test_name+"-jh"
```
Comment (piobig2871) [inline features/steps/tools.py:189]: fix spaceing

File: features/steps/tools.py (hunk 1 above)
Comment (fraugabel) [reply features/steps/tools.py:182]: i don't really see whats wrong there? the identation is matching the rest

File: features/steps/definitions.py (hunk 6 above)
Comment (fraugabel) [reply features/steps/definitions.py:555]: thanks, didn't see that solution

**Review by piobig2871 — APPROVED** [review]

**Pull request merged**